tests/test-dirs/destruct/prefixing.t

```diff
@@ -0,0 +1,108 @@
+#################
+### FROM EXPR ###
+#################
+Test 1.1
+  $ $MERLIN single case-analysis -start 3:2 -end 3:3 -filename typ.ml <<EOF | \
+  > tr -d '\n' | jq '.value[1]'
+  > module A = struct type my_list = Atom | Elt of string * my_list end
+  > let f x : A.my_list =
+  >   x
+  > EOF
+  "match (x : A.my_list) with | A.Atom -> (??) | A.Elt (_, _) -> (??)"
```

Comment:
how did the type annotation on `x` appear?

I know that's unrelated by the content of the PR, but I'm somewhat baffled (also, isn't it incorrect?)

Reply:
I have no idea, but it is valid.

Reply:
Is it? Nothing in the source indicates that `x` has that type, that type is the return type of the function.

Reply:
But the body of the function is the single expression `x` ?

Reply:
So the return type of the function is the same as the type of `x`.

Reply:
Right, I guess that's fair.
Still surprised that an annotation is added, but that's a mystery for another time.